virttest/test_setup.py

```diff
@@ -1362,10 +1375,12 @@ def remove_driver(self, driver=None):
         if ARCH == 'ppc64le' and driver == 'mlx5_core':
             pf_devices = self.get_pf_mlx()
             logging.info("Mellanox PF devices '%s'", pf_devices)
-            for PF in pf_devices:
-                if not self.set_vf(PF):
-                    return False
             cmd = "rmmod mlx5_ib;modprobe -r mlx5_core;modprobe mlx5_ib"
+        elif ARCH == 'x86_64':
+            pf_devices = self.get_pf_ids()
+        for PF in pf_devices:
+            if not self.set_vf(PF):
+                return False
```

Comment:
@huangyum, In Older code you can see `set_vf()` is performed before removing mlx drivers for ppc64le, with this change the mlx5 drivers will be removed in L1378 after which `set_vf()` will fail.

Reply:
Thanks for your review. L1378 only sets 'cmd', which will be executed later in L1384.  set_vf() is executed before that so it would not fail.

Reply:
Got it, Thanks!